packages/apis/src/interfaces/tasks.rs

```diff
@@ -2,16 +2,47 @@
 /// by other contracts. This specifies the required APIs for on-chain interactions.
 /// This must be a subset of any of the implementation.
 use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Addr, StdResult, Storage};
 // FIXME: do we need to derive cw_orch? This is not used by the contracts calling,
 // And the external callers likely use the full APIs.
 // TODO: try embedding these (enum with serde(untagged)) in the main types of the implementing contracts
 use cw_orch::{ExecuteFns, QueryFns};
+use cw_storage_plus::Map;
 
-use crate::{id::TaskId, tasks::Status};
+use crate::{id::TaskId, tasks::Status, verifier_simple::TaskMetadata};
 
 // FIXME: hot to make these generic
 pub type ResponseType = serde_json::Value;
 
+pub struct TasksStorage<'a>(Map<(&'a Addr, TaskId), TaskMetadata>);
+
+impl<'a> TasksStorage<'a> {
```

Comment:
I was thinking to add `handle_metadata` here too, but fine without that for now.